src/drivers/nuttx/lv_nuttx_fbdev.c

```diff
@@ -138,21 +139,33 @@ int lv_nuttx_fbdev_set_file(lv_display_t * disp, const char * file)
     uint32_t w = dsc->vinfo.xres;
     uint32_t h = dsc->vinfo.yres;
     uint32_t stride = dsc->pinfo.stride;
-    uint32_t data_size = h * stride;
+    size_t data_size = h * stride;
     lv_draw_buf_init(&dsc->buf1, w, h, color_format, stride, dsc->mem, data_size);
 
-    /* double buffer mode */
-
+    /* Check buffer mode */
     bool double_buffer = dsc->pinfo.yres_virtual == (dsc->vinfo.yres * 2);
     if(double_buffer) {
         if((ret = fbdev_init_mem2(dsc)) < 0) {
             goto errout;
         }
 
         lv_draw_buf_init(&dsc->buf2, w, h, color_format, stride, dsc->mem2, data_size);
+        lv_display_set_draw_buffers(disp, &dsc->buf1, &dsc->buf2);
+    }
+    else {
+        dsc->mem_off_screen = malloc(data_size);
```

Comment:
```suggestion
        dsc->mem_off_screen = lv_malloc(data_size);
```

lv_malloc?

Reply:
Users may use a separate heap for lvgl, which is usually smaller. The SDL porting also uses the drawing buffer allocated by libc.

Reply:
Confirmed, for these huge allocations `malloc` is preferred. `lv_malloc` is more for widgets, styles, etc.